components/filter/filter.js

```diff
@@ -544,7 +549,16 @@ class Filter extends LocalizeCoreElement(RtlMixin(LitElement)) {
 
 		if (shouldSearch) this._performDimensionSearch(dimension);
 		if (shouldRecount) this._setFilterCounts(dimension);
-		if (shouldUpdate) this.requestUpdate();
+		if (shouldUpdate)  {
+			this.requestUpdate();
+
+			const singleDimension = this._dimensions.length === 1;
+			if (singleDimension && !this._waitingForResize) {
+				this._waitingForResize = true;
+				const dropdown = this.shadowRoot.querySelector('d2l-dropdown-content');
+				dropdown.renderOnResize(this.shadowRoot.querySelector('#d2l-filter-container'));
```

Comment:
We only need to call this once - when filters are initially loading into a single dimension filter.

Reply:
With HM is might get a bit more choppy with the loading, versus the demo page which is all loaded at the same time.  I'm thinking something like this would be good:
```
			} else if (prop === 'values') {
				if (dimension.searchValue) shouldSearch = true;
				shouldRecount = true;
				shouldResizeDropdown = true;
				this._activeFiltersSubscribers.updateSubscribers();
			} else if (prop === 'loading') {
				shouldResizeDropdown = true;
			}
		});

		if (shouldSearch) this._performDimensionSearch(dimension);
		if (shouldRecount) this._setFilterCounts(dimension);
		if (shouldUpdate) this.requestUpdate();
		if (shouldResizeDropdown) {
			if (this._dimensions.length === 1 && this.opened) {
				const dropdown = this.shadowRoot.querySelector('d2l-dropdown-content');
				const container= this.shadowRoot.querySelector('.d2l-filter-container');
				if (container) dropdown.renderOnResize(container);
		}
```
Then it only runs on relevant property updates (new values or changing loading states) and can run again if it goes back to loading state or if more options pop in.